File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int height();
void builder(int size);

int main(void)
{
    int h = height();
    builder(h);
}


//function acquires desired build height
int height()
{
    int x;
    do
    {
        x = get_int("Height: ");
    }
    while (x < 1 || x > 8);
    return x;
}
//function builds pyramid
void builder(int size)
{
    //declaring prototype values for later use
    int space = 0;
    int width = 1;
    int blocks = 0;
    //executes builder until desired height
    do
    {
        //prints spaces for pyramid
        for (space = 0; space < size - width; space++)
        {
            printf(" ");
        }

        //left side build
        for (blocks = 0; blocks < width; blocks++)

        {
            printf("#");
        }

        //adds gap
        printf("  ");

        //right side build
        for (blocks = 0; blocks < width; blocks++)
        {
            printf("#");
        }
        printf("\n");
        width++;
    }
    while (width <= size);
}

File: sentimental-mario-more/mario.py

def height():
    x = 0
    # requests height from user
    # loops until user inputs valid height
    while (x < 1 or x > 8):
        # handles non-integers
        try:
            x = int(input("Height: "))
        except ValueError:
            print("Not a number my man, try again")

    return x
# builds pyramid according to size input


def builder(size):
    space = 0
    width = 1
    blocks = 0

    while width <= size:
        # prints spaces for sides
        for space in range(size - width):
            print(" ", end="")
        # left side of build
        for blocks in range(width):
            print("#", end="")
        # prints space in the middle
        print("  ", end="")
        # right side of build
        for blocks in range(width):
            print("#", end="")
        print()
        width += 1


def main():
    h = height()
    builder(h)
    

main()